### PR TITLE
Fixup system - follow up on #4356

### PR DESCRIPTION
**Pull request by timse**
followup on #4356

see there for detailed description

### CODE REVIEW

File: test/cases/parsing/issue-2942/index.js [hunk 1]
```diff
@@ -1,5 +1,5 @@
 it("should polyfill System", function() {
-	if (typeof System === "object" && typeof System.register === "function") {
+	if (typeof System.register === "function") {
```
Comment (sokra) [inline test/cases/parsing/issue-2942/index.js]: Why are changes to the tests necessary?

File: lib/ParserHelpers.js [hunk 2]
```diff
@@ -3,64 +3,75 @@
...
-				dep.userRequest = name;
-				deps.push(dep);
-			}
-		},
-		module: parser.state.module
-	});
-	parser.state.current.addVariable(name, expression, deps);
-	return true;
-};
+module.exports = class ParserHelpers {
```
Comment (sokra) [inline lib/ParserHelpers.js]: ParserHelpers is not a real class. Stay with the `exports.addParsedVaraibleToModule` style. It's more aligned with the future style `export function addParsedVariableToModule()`.

File: test/cases/parsing/typeof/index.js [hunk 3]
```diff
@@ -45,7 +45,6 @@ it("should not parse filtered stuff", function() {
 	if(typeof module === "undefined") module = require("fail");
 	if(typeof module != "object") module = require("fail");
 	if(typeof exports == "undefined") exports = require("fail");
-	if(typeof System !== "object") exports = require("fail");
```
Comment (sokra) [inline test/cases/parsing/typeof/index.js]: Why are changes to the tests necessary?

`typeof System` should `object` in webpack.

**Comment by timse** [comment]
reverted the refactor. 
need to check again why i needed to change the `typeof checks` i think i had a reason :)

**Comment by timse** [comment]
changes are reverted. 
thanks @sokra for taking time with this 👍

File: buildin/system.js [hunk 4]
```diff
@@ -1,3 +1,6 @@
-/* global System */
 // "fix" for users of "System" global
```
Comment (sokra) [inline buildin/system.js]: fix comment

**Comment by sokra** [comment]
Thanks

**Pull request merged**